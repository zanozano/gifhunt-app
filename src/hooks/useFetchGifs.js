import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
	//* useState
	const [state, setState] = useState({
		data: [],
		loading: true,
	});

	//* useEffect
	useEffect(() => {
		getGifs(category).then((imgs) => {
			setState({
				data: imgs,
				loading: false,
			});
		});
	}, [category]);

	return state;
};
