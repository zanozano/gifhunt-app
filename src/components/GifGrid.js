import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
	// Renombre la data para usar como images
	//*
	const { data: images, loading } = useFetchGifs(category);

	return (
		<>
			<h2 className='animate__animated animate__fadeIn'>{category}</h2>
			{loading && <p className='animate__animated animate__flash'>Loading</p>}
			<div className='card-grid'>
				{images.map((img) => (
					<GifGridItem
						key={img.id}
						//Se esta mandando el obj spread directamente
						{...img}
					/>
				))}
			</div>
		</>
	);
};
GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};

export default GifGrid;
