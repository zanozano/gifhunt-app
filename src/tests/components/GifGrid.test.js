import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifGrid from '../../components/GifGrid';

import { useFetchGifs } from '../../hooks/useFetchGifs';
// Se esta simulando la respuesta de la API
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/> ==>', () => {
	//
	const category = 'One Punch';

	test('Prueba de snapshot', () => {
		// Se esta falseando el fetch
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});

		//
		const wrapper = shallow(<GifGrid category={category} />);
		expect(wrapper).toMatchSnapshot();
	});
	test('Debe de mostrar items cuando se cargar el useFetchGift', () => {
		const gifs = [
			{
				id: 'ABC',
				title: 'Cualquier cosa',
				url: 'https//localhost/cualquiercosa/cosa.jpg',
			},
			{
				id: 'CRD',
				title: 'Cualquier cosa 2',
				url: 'https//localhost/cualquiercosa/cosa.jpg',
			},
		];

		// Se esta falseando el fetch
		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});

		const wrapper = shallow(<GifGrid category={category} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('p').exists()).toBe(false);
		expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
	});
});
