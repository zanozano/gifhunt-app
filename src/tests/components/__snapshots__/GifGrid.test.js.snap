// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> ==> Debe de mostrar items cuando se cargar el useFetchGift 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https//localhost/cualquiercosa/cosa.jpg"
    />
    <GifGridItem
      id="CRD"
      key="CRD"
      title="Cualquier cosa 2"
      url="https//localhost/cualquiercosa/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid/> ==> Prueba de snapshot 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    One Punch
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
